crates/youki/src/workload/wasmedge.rs

```diff
@@ -1,19 +1,18 @@
-use anyhow::Result;
 use oci_spec::runtime::Spec;
 use wasmedge_sdk::{
     config::{CommonConfigOptions, ConfigBuilder, HostRegistrationConfigOptions},
     params, VmBuilder,
 };
 
-use libcontainer::workload::Executor;
+use libcontainer::workload::{Executor, ExecutorError};
 
 const EXECUTOR_NAME: &str = "wasmedge";
 
 #[derive(Default)]
 pub struct WasmEdgeExecutor {}
 
-impl Executor for WasmEdgeExecutor {
-    fn exec(&self, spec: &Spec) -> Result<()> {
+impl WasmEdgeExecutor {
+    fn exec_inner(spec: &Spec) -> anyhow::Result<()> {
```

Comment:
Why don't we use this error here?

Reply:
This is mostly to minimize the changes since the original code uses anyhow. The implementation of the 3 wasm executor is implemented in `youki`, not `libcontainer`, so there is no hard requirement to remove `anyhow`. I was not sure how best to approach this so I wrapped the original implementation into the `exec_inner` and only `map_err` once to meet the trait requirement of `Executor`.

I also wasn't sure how best to implement the `Executor` trait and the interface `libcontainer` should provide. This PR does the minimal so there are still rooms for improvement. I have a vague plan to implement the `Executor` trait with an `Error` associated type, so the implementation can provide whatever error type they wish. However, this proves to be tricky because we need to store the `Executor` in `ContainerBuilder` and `ContainerArgs`. I could not figure out how to store a trait object with unknown associated type yet. Perhaps the new Generic Associated Type can be of some help. In the end, I decided to keep things simple for now to get this PR merged, and then we can figure out how best to approach this.

On a different note, I suspect that `Executor` interface can become a great thing for youki and libcontainer, because it opens up a lot of possibility. Currently, we switching the the default `execvp` calls for the wasm calls. I can also switching the default `execvp` to run an arbitrary rust function. This would allow us to better and easier testing the `libcontainer` directly, without the full setup of the oci bundle.

This is a long winded way to answer your question :) But I can certainly remove the `anyhow` from these executor as well.

Reply:
Done. Fixed these wasm executor not using anyhow.